packages/otel/src/otel-urql-exchange-factory.ts

```diff
@@ -0,0 +1,123 @@
+import { context, Span, SpanKind, SpanStatusCode, Tracer } from "@opentelemetry/api";
+import { SemanticAttributes } from "@opentelemetry/semantic-conventions";
+import { type CombinedError, makeOperation, mapExchange, Operation } from "urql";
+
+type Definition = {
+  name: {
+    value: string;
+  };
+};
+
+interface ExtendedFetchOptions extends RequestInit {
+  headers: Record<string, string>;
+}
+
+type ExtendedOperationContext = Operation["context"] & {
+  span: Span;
+  fetchOptions?: ExtendedFetchOptions;
+};
+
+interface ExtendedOperation extends Operation {
+  context: ExtendedOperationContext;
+}
+
+const GraphQLAttributeNames = {
+  OPERATION_TYPE: "graphql.operation.type",
+  OPERATION_NAME: "graphql.operation.name",
+  OPERATION_BODY: "graphql.operation.body",
+  OPERATION_KEY: "graphql.operation.key",
+  VARIABLES: "graphql.variables.",
+} as const;
+
+export const createOtelUrqlExchange = (args: { tracer: Tracer }) => {
```

Comment:
maybe different scope but you can check if there are any native graphql instrumentations (probably something that will parse http payload)?

AFAIR we have this custom operation because we didn;t make it work:
https://www.npmjs.com/package/@opentelemetry/instrumentation-graphql

Reply:
I've plan to do this in the next PR